rbc/omnisci_array.py

```diff
@@ -181,17 +235,30 @@ def impl(a):
 
 
 @extending.overload(np.prod)
-def omnisci_np_prod(a):
+@extending.overload_method(ArrayPointer, 'prod')
+def omnisci_np_prod(a, initial=None):
     if isinstance(a, ArrayPointer):
-        def impl(a):
-            s = 1
+        def impl(a, initial=None):
+            if initial is not None:
+                s = initial
+            else:
+                s = 1
             n = len(a)
             for i in range(n):
                 s *= a[i]
             return s
         return impl
 
 
+@extending.overload(np.mean)
+@extending.overload_method(ArrayPointer, 'mean')
+def omnisci_array_mean(x):
+    if isinstance(x, ArrayPointer):
+        def impl(x):
+            return sum(x) / len(x)
```

Comment:
Check if `len(x) == 0`. If True, report `Mean of empty array` via printf or exception, and return `nan` (for floats) or `0` for ints. Ideally, we should raise an exception though..